cmd/experimental/kjobctl/apis/v1alpha1/job_template_types.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// JobTemplateSpec describes the data a job should have when created from a template
+type JobTemplateSpec struct {
+	// Standard object's metadata.
+	//
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of the job.
+	//
+	// +kubebuilder:validation:Required
+	Spec v1.JobSpec `json:"spec"`
+}
+
+// +genclient
+// +kubebuilder:object:root=true
+
+// JobTemplate is the Schema for the jobtemplate API
+type JobTemplate struct {
```

Comment:
One more or less hacky alternative to this wold be to use a [CronJob](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/cron-job-v1/#CronJobSpec) that is kept suspended. @alculquicondor @mwielgus WDYT?

Reply:
Let's not do that.